Extractor/History/HistoryReader.cs

```diff
@@ -276,31 +278,39 @@ private void RestartHistoryInStates()
 
         public void RemoveIssue(StateIssue issue)
         {
-            bool stateOk = state.IsGood;
-
-            if (state.RemoveIssue(issue))
+            lock (statesLock)
             {
-                log.LogDebug("Removed history issue: {Issue}. New state: {State}", issue, state);
-            }
+                bool stateOk = state.IsGood;
 
-            if (stateOk != state.IsGood)
-            {
-                stateChangedEvent.Set();
+                if (state.RemoveIssue(issue))
+                {
+                    log.LogDebug("Removed history issue: {Issue}. New state: {State}", issue, state);
+                }
+
+                if (stateOk != state.IsGood)
+                {
+                    log.LogDebug("History state changed, history will {Event}", state.IsGood ? "Restart" : "Stop");
```

Comment:
Maybe we can just hardcode Restart here, since it should never say Stop.

Reply:
It can, the condition is not a check of `state.IsGood`, it's a check of whether `state.IsGood` has changed, it goes both ways.

Reply:
The method calls RemoveIssue, and that can only change IsGood from false to true. This block is dependent on there being a change, therefore this can only say Restart.

Reply:
Ah. You're right.